apps/server-discover/main.c

```diff
@@ -262,6 +299,9 @@ int main(int argc, char *argv[])
             BACNET_MAX_INSTANCE);
         return 1;
     }
+    if (specific_address) {
+        bacnet_address_init(&dest, &mac, dnet, &adr);
```

Comment:
dnet could be -1 here. please add a check for dnet in range and assign to broadcast if out of range by default.  for example:
```
        if ((dnet < 0) || (dnet > BACNET_BROADCAST_NETWORK)) {
            dnet = BACNET_BROADCAST_NETWORK;
        }
        bacnet_address_init(&dest, &mac, dnet, &adr);
```

Reply:
`--dnet` actually had two branches in the command line parsing loop, so I removed one.

The other was already bounds-checking the result of `strtol`, which allows `dnet` to be initialized with a valid default of `BACNET_BROADCAST_NETWORK` so the bounds check doesn't need to be repeated at point of use.